driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -720,7 +719,7 @@ boolean checkSchemaAgreement() throws ConnectionException, BusyConnectionExcepti
         if (connection == null || connection.isClosed())
             return false;
 
-        DefaultResultSetFuture peersFuture = new DefaultResultSetFuture(null, cluster.protocolVersion(), new Requests.Query(SELECT_SCHEMA_PEERS));
+        DefaultResultSetFuture peersFuture = new DefaultResultSetFuture(null, cluster.protocolVersion(), new Requests.Query(SELECT_PEERS));
```

Comment:
This fixes things, but wasn't sure if it was the best solution, as we only want to consider peers that are valid, which requires retrieving all of the peer rows.  Alternatively we could skip peers that we don't have in host metadata, although that may be more involved / less reliable.